.github/workflows/build-ubuntu.yaml

```diff
@@ -0,0 +1,59 @@
+name: build-ubuntu
+
+
+# Controls when the action will run
+on:
+  push:
+  pull_request:
+  workflow_dispatch: # trigger the workflow manually
+  # # trigger the workflow on schedule
+  # # https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#scheduled-events
+  # schedule:
+  #   - cron:  '* * * * *'
+
+
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  build:
+    runs-on: ${{ matrix.runner }}
+    continue-on-error: false
+    strategy:
+      matrix:
+        runner: [ubuntu-16.04, ubuntu-18.04, ubuntu-20.04]
+        compiler: [clang++, g++]
+        python: [python2, python3]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install requirements
+        run: >
+          sudo apt install --no-install-recommends -y
+          build-essential
+          clang
+          python
+          python-numpy
+          python3
+          python3-numpy
+          cmake
+          gfortran
+          git
+          libopenmpi-dev
+          openmpi-bin
+          openssh-client
+      - name: Build
+        run: >
+          rm -rf build && mkdir -p build && cd build &&
+          cmake
+          -D CMAKE_INSTALL_PREFIX=./TasmanianInstall
+          -D CMAKE_CXX_FLAGS="-O3 -Wall -Wextra -Wshadow -pedantic"
+          -D CMAKE_CXX_COMPILER=${{ matrix.compiler }}
+          -D Tasmanian_ENABLE_PYTHON=ON
+          -D PYTHON_EXECUTABLE=/usr/bin/${{ matrix.python }}
+          -D Tasmanian_ENABLE_FORTRAN=ON
+          -D Tasmanian_TESTS_OMP_NUM_THREADS=4 .. &&
+          make -j4
+      - name: Test
+        run: cd build && ctest -j4 -V --no-compress-output -T Test
```

Comment:
`ctest -j4` will launch 4 tests simultaneously while `Tasmanian_TESTS_OMP_NUM_THREADS=4` will launch 4 threads per test. That's total of 16 threads.

Just making sure this is what you intended.